src/vehicle.cpp

```diff
@@ -1648,6 +1648,19 @@ bool vehicle::can_mount(int const dx, int const dy, const vpart_id &id) const
         }
     }
 
+    //Turrets must be installed on a turret mount
+    if( part.has_flag( "TURRET" ) ) {
+        bool anchor_found = false;
+        for( const auto &elem : parts_in_square ) {
+            if( part_info( elem ).has_flag( "TURRET_MOUNT" ) ) {
+                anchor_found = true;
```

Comment:
break;

Reply:
There is no `break;`s in similar snippets above and below here in `vehicle.cpp`.

Reply:
Well in this specific case its pretty clear that once an anchor is found additional parts don't need to be checked for the TURRET_MOUNT flag. The only possible scenario after setting anchor_found to true is setting anchor_mount to true again, which it already is, so a no-op. It's more efficient to stop checking than to do further processing that can't change anything.

Other parts in the file do it properly, just to pick a few:
https://github.com/CleverRaven/Cataclysm-DDA/blob/bb912354e10d7e8ea510de335b6572100ec2634a/src/vehicle.cpp#L206
https://github.com/CleverRaven/Cataclysm-DDA/blob/bb912354e10d7e8ea510de335b6572100ec2634a/src/vehicle.cpp#L731
https://github.com/CleverRaven/Cataclysm-DDA/blob/bb912354e10d7e8ea510de335b6572100ec2634a/src/vehicle.cpp#L5380